src/deploymentModals.jsx

```diff
@@ -0,0 +1,213 @@
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+
+import cockpit from 'cockpit';
+import client from './client';
+
+import { Alert } from "@patternfly/react-core/dist/esm/components/Alert";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Checkbox } from '@patternfly/react-core/dist/esm/components/Checkbox';
+import { Modal } from "@patternfly/react-core/dist/esm/components/Modal";
+import { Stack } from "@patternfly/react-core/dist/esm/layouts/Stack";
+import { Text } from "@patternfly/react-core/dist/esm/components/Text";
+
+import { useDialogs } from "dialogs.jsx";
+
+const _ = cockpit.gettext;
+
+export const CleanUpModal = ({ os }) => {
+    const Dialogs = useDialogs();
+
+    const [deleteTemporaryFiles, setDeleteTemporaryFiles] = useState(true);
+    const [deleteRPMmetadata, setDeleteRPMmetadata] = useState(true);
+    const [deletePendingDeployments, setDeletePendingDeployments] = useState(false);
+    const [deleteRollbackDeployments, setDeleteRollbackDeployments] = useState(false);
+    const [buttonLoading, setButtonLoading] = useState(false);
+    const [error, setError] = useState("");
+
+    const doCleanup = () => {
+        setButtonLoading(true);
+        setError("");
+
+        const cleanupFlags = [];
+        if (deleteTemporaryFiles) {
+            cleanupFlags.push("base");
+        }
+        if (deleteRPMmetadata) {
+            cleanupFlags.push("repomd");
+        }
+        if (deletePendingDeployments) {
+            cleanupFlags.push("pending-deploy");
+        }
+        if (deleteRollbackDeployments) {
+            cleanupFlags.push("rollback-deploy");
+        }
+
+        return client.run_transaction("Cleanup", [cleanupFlags], os)
+                .then(Dialogs.close)
+                .catch(ex => {
+                    console.warn(ex);
+                    setError(ex.message);
+                    setButtonLoading(false);
+                });
+    };
+
+    const actions = [
+        <Button key="cleanup-accept"
+            variant="primary"
+            isDisabled={buttonLoading || (!deleteTemporaryFiles && !deleteRPMmetadata && !deletePendingDeployments && !deleteRollbackDeployments)}
+            isLoading={buttonLoading}
+            onClick={() => doCleanup()}
+        >
+            {_("Clean up")}
+        </Button>,
+        <Button key="cleanup-cancel" variant="link" onClick={Dialogs.close}>
+            {_("Cancel")}
+        </Button>
+    ];
+
+    return (
+        <Modal isOpen
+            id="cleanup-deployment-modal"
+            title={_("Remove temporary files")}
+            position="top"
+            variant="small"
+            onClose={Dialogs.close}
+            actions={actions}
+        >
+            {error &&
+                <Alert variant="danger"
+                    isInline
+                    title={error}
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231128-135012-30a3544e-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)